### PR TITLE
feat: add RGB to HSL color format conversion algorithm

### PR DESCRIPTION
**Pull request by piyushk77**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Conversions/RgbHslConversion.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @example
+ * const colorRgb = [24, 98, 118]
+ *
+ * const result = rgbToHsl(colorRgb)
+ *
+ * // The function returns the corresponding color in HSL format:
+ * // result = [193, 66, 28]
+ */
+
+const checkRgbFormat = (colorRgb) => {
```
Comment (appgurueu) [inline Conversions/RgbHslConversion.js:18]: Could write this concisely as `colorRgb => colorRgb.map(c => c >= 0 && c <= 255).every()`

**Review by appgurueu — CHANGES_REQUESTED** [review]
Has the same issue as the other PR (`return "error string"` instead of `throw`). Again this should be made to use a `throw`, and the tests should be partitioned accordingly. Note that you can use `expect(...).toThrowError("error message")`. Otherwise looks fine!

File: Conversions/RgbHslConversion.js (hunk 1 above)
Comment (piyushk77) [reply Conversions/RgbHslConversion.js:18]: Got it.

**Review by piyushk77 — COMMENTED** [review]
@appgurueu I have made the necessary changes. Please have a look.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**